vips/image.go

```diff
@@ -654,20 +654,24 @@ func (r *ImageRef) Invert() error {
 	return nil
 }
 
-// Resize executes the 'resize' operation
 func (r *ImageRef) Resize(scale float64, kernel Kernel) error {
-	err := r.PremultiplyAlpha()
+	out, err := vipsResize(r.image, scale, kernel)
 	if err != nil {
 		return err
 	}
+	r.setImage(out)
 
-	out, err := vipsResize(r.image, scale, kernel)
+	return nil
+}
+
+func (r *ImageRef) AlphaResize(scale float64) error {
```

Comment:
Please don't expose the alpha details to the outside, in the Resize function check for alpha and call the proper **private** function accordingly

Reply:
When doing the resize on images with alpha the algorithm parameter is ignored, I don't want govips interface to lie

Reply:
I rather lie and add the algo support later on (if needed...), getting rid of public function is pain.

The only reason the the algo argument is mandatory and not optional is that Go does not support overloading function

Reply:
ok

Reply:
done